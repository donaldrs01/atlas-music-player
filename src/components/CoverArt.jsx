import React from 'react';
import placeholder from "../assets/placeholder.svg"

const CoverArt = () => {
    return (
        <div className=" w-full bg-gray-200 flex items-center justify-center aspect-square mt-4">
            <img
                src={placeholder}
                className="w-full h-full object-cover"
            />
        </div>
    );
};

export default CoverArt;