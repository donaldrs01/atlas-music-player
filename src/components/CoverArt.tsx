{/*
import React from 'react';
import placeholder from "../assets/album_art.jpg"

const CoverArt = () => {
    return (
        <div className="w-full max-h-[70%] bg-gray-200 flex items-center justify-center aspect-square mb-4">
            <img
                src={placeholder}
                className="w-full h-full object-cover"
            />
        </div>
    );
};

export default CoverArt;
*/}
import React from 'react';

{/*
// Type define different song parameters
interface SongData {
    id: string;
    title: string;
    artist: string;
    genre: string;
    duration: number;
    cover: string;
    song: string;
}
    */}

// Type define prop for CoverArt (songID to fetch art)
interface CoverArtProps {
    coverUrl: string;
}

const CoverArt: React.FC<CoverArtProps> = ({ coverUrl }) => {
    return (
        <div className="w-full max-h-[70%] bg-gray-200 flex items-center justify-center aspect-square mb-4">
            <img
                src={coverUrl || '/assets/placeholder.svg'}
                alt="Album Cover"
                className="w-full h-full object-cover"
            />
        </div>
    );
};

export default CoverArt;

/*
const CoverArt: React.FC<CoverArtProps> = ({ songId }) => {
    const [coverArt, setCoverArt] = useState<string>('');
    const [loading, setLoading] = useState<boolean>(true);
    const [error, setError] = useState<string | null>(null); // Can either be string or null depending on existence of error

    useEffect(() => {
        const fetchCoverArt = async() => {
            try {
                const response = await fetch(`/api/v1/songs/${songId}.json`);
                if (!response.ok) {
                    throw new Error('Failed to fetch song data');
                }

                const data: SongData = await response.json();
                setCoverArt(data.cover);
            } catch (err) {
                setError('Error fetching album cover art');
            } finally {
                setLoading(false);
            }
        };

        fetchCoverArt();
    }, [songId]);

    if (loading) return <div className="text-gray-400">Loading cover art...</div>;
    if (error) return <div className="text-red-700">{error}</div>;

    return (
        <div className="w-full max-h-[70%] bg-gray-200 flex items-center justify-center aspect-square mb-4">
            <img
                src={coverArt}
                alt="Album Cover"
                className="w-full h-full object-cover"
            />
        </div>
    );
};

export default CoverArt;
*/