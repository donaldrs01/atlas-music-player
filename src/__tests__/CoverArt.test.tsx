import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import CoverArt from "../components/CoverArt";

describe("CoverArt Snapshot Tests", () => {
  test("displays generic image", () => {
    const { container } = render(
      <CoverArt coverUrl="https://utfs.io/f/E9fJnaKtTy1bOclGsk1hB7xMLwUVFDiXypZukQrcnYbgdEv6" />,
    );
    expect(container).toMatchSnapshot();
  });

  test("displays another album cover", () => {
    const { container } = render(
      <CoverArt coverUrl="https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp" />,
    );
    expect(container).toMatchSnapshot();
  });
});
