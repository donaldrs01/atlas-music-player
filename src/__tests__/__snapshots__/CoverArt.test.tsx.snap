// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Snapshot Tests > displays another album cover 1`] = `
<div>
  <div
    class="flex h-full w-full justify-center"
  >
    <img
      alt="Album Cover"
      class="h-[30vh] w-full max-w-[95%] object-cover md:h-[55vh]"
      src="https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp"
    />
  </div>
</div>
`;

exports[`CoverArt Snapshot Tests > displays generic image 1`] = `
<div>
  <div
    class="flex h-full w-full justify-center"
  >
    <img
      alt="Album Cover"
      class="h-[30vh] w-full max-w-[95%] object-cover md:h-[55vh]"
      src="https://utfs.io/f/E9fJnaKtTy1bOclGsk1hB7xMLwUVFDiXypZukQrcnYbgdEv6"
    />
  </div>
</div>
`;
